src/hotspot/cpu/x86/templateInterpreterGenerator_x86.cpp

```diff
@@ -713,23 +713,57 @@ address TemplateInterpreterGenerator::generate_Reference_get_entry(void) {
 }
 
 void TemplateInterpreterGenerator::bang_stack_shadow_pages(bool native_call) {
```

Comment:
The watermark algorithm should also work on other architectures such as aarch64, right?

Reply:
Yes, as I stated in PR text: "This patch makes a pilot change for x86, without touching other architectures. Other architectures can follow this example later. "